src/instruments/abstract_instruments/instrument.py

```diff
@@ -430,12 +442,14 @@ def open_from_uri(cls, uri):
             raise NotImplementedError("Invalid scheme or not yet " "implemented.")
 
     @classmethod
-    def open_tcpip(cls, host, port):
+    def open_tcpip(cls, host, port, username=None, password=""):
```

Comment:
Just an idea, maybe it would be useful to adopt the approach `requests` takes:
an `auth=` keyword argument which can hold different things.
For example
* `auth=("user", "password")`
* [some custom auth](https://requests.readthedocs.io/en/latest/user/advanced/#custom-authentication)

https://requests.readthedocs.io/en/latest/user/authentication/

Some class-based auth is probably overkill for the moment, but using a single argument would allow flexibility in the future (for example if API keys could be used, or reading from a file).

Reply:
Thanks for the comment, I quite like the idea. That would make it much more flexible and the `auth` keyword could hold different authentication arguments for different instruments, i.e., whatever is required. Will implement that.